instrumentation/opentelemetry-instrumentation-django/src/opentelemetry/instrumentation/django/middleware.py

```diff
@@ -12,22 +12,22 @@
 # See the License for the specific language governing permissions and
 # limitations under the License.
 
-import time
 from logging import getLogger
+from time import time
 
 from django.conf import settings
 
-from opentelemetry.configuration import Configuration
 from opentelemetry.context import attach, detach
 from opentelemetry.instrumentation.django.version import __version__
 from opentelemetry.instrumentation.utils import extract_attributes_from_object
-from opentelemetry.instrumentation.wsgi import (
```

Comment:
I believe this wsgi instrumentation library could be used for any wsgi framework to collect telemetry. For instance let's say I have a WSGI based framework X and there is no standalone instrumentation for it. If we remove this what would be the way for collecting telemetry?

Reply:
The WSGI code as it is now is not an instrumentation because it does no implement `BaseInstrumentor`. There is no WSGI library that is instrumented by it as opposed to specific libraries like Flask, Django, etc. For that reason, the WSGI code should not be in `opentelemetry.instrumentation`.

I don't understand what you mean with _"For instance let's say I have a WSGI based framework X and there is no standalone instrumentation for it. If we remove this what would be the way for collecting telemetry?"_, can you explain, please?

Reply:
The [WSGI](https://opentelemetry-python.readthedocs.io/en/stable/instrumentation/wsgi/wsgi.html) and [ASGI](https://opentelemetry-python.readthedocs.io/en/stable/instrumentation/asgi/asgi.html) are different instrumentation libraries on their own as of now. I may be wrong but my understanding is that I can use [opentelemetry-instrumentation-wsgi](https://opentelemetry-python.readthedocs.io/en/stable/instrumentation/wsgi/wsgi.html#module-opentelemetry.instrumentation.wsgi) with any wsgi based web frameworks to [traces](https://github.com/open-telemetry/opentelemetry-python-contrib/blob/master/instrumentation/opentelemetry-instrumentation-wsgi/src/opentelemetry/instrumentation/wsgi/__init__.py#L203). I will try to use an example to clarify the last part, We don't have the instrumentation library for [cherrpy](https://docs.cherrypy.org/en/latest/index.html) framework but I can still instrument my `cherrpy` based application using `opentelemetry-instrumentation-wsgi`. It would look something similar as followed
```python
import cherrypy
from opentelemetry.instrumentation.wsgi import OpenTelemetryMiddleware

class HelloWorld(object):
    @cherrypy.expose
    def index(self):
        return "Hello World!"

app  = cherrypy.Application(HelloWorld(), '/')

app.wsgiapp = OpenTelemetryMiddleware(app.wsgiapp)

cherrypy.quickstart(app)
```
I didn't think of WSGI/ASGI `OpenTelemetryMiddleware` as an util although there are some util functions in that package. It  was easier to find that library, docs and use because it is very similar to any other popular middlewares [flask-cache](https://pythonhosted.org/Flask-Cache/) or [bottle](https://bottlepy.org/docs/dev/recipes.html) middlewares etc.. to name few. I thought may be util is not an ideal place to put middleware kind of stuff, just wanted to understand the motivation behind moving from individual packages to util. On the other hand I think it is good to move the helper functions under `util.http`.